DSESwift/DataSyncer.swift

```diff
@@ -255,6 +275,25 @@ class DataSyncer : NSObject {
         } while (!isCompleted)
     }
     
+    private func getSlicedArray(array: Array<DataPoint>,startIndex: Int) -> Array<DataPoint>{
+        // endIndex: remaining length from startIndex or next startIndex
+        let endIndex = min(array.count - startIndex, (startIndex + 1) * DSEConstants.DEFAULT_REMOTE_POST_LIMIT)
+        return Array(array[startIndex ..< endIndex])
+    }
+    
+    private func insertOrUpdateSensorIntoLocal(sensor: Sensor) throws{
+        if try !DatabaseHandler.hasSensor(sensor.source, sensorName: sensor.name) {
+            do{
+                try DatabaseHandler.insertSensor(sensor)
+                self.delegate?.onSensorCreated(sensor.name)
+            }catch DSEError.InvalidSensorName {
```

Comment:
DSE does not know about valid source names at the moment. Does Android check the source name on the mobile side?